### PR TITLE
1223 R-ops UI Link a QID to a case

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context
We want to be able to link QID's to cases from the R-ops UI.

# What has changed
* Add link box to the case details page
* Add QID link confirmation page
* Add flash messages to confirm a link has been submitted or error if the QID does not exist

# How to test?
Build and run the AT's branch

# Links
https://trello.com/c/6QdthBHC/1223-r-ops-ui-link-a-qid-to-a-case-13

### CODE REVIEW

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Requires some audit logging that can be easily extracted/searched in the logs

File: census_rm_ops_ui/controllers/case_controller.py [hunk 1]
```diff
@@ -33,3 +33,31 @@ def get_all_case_details(case_id, case_api_url):
...
+        logger.error('Error searching for qid', qid=qid)
+        raise
+
+    return response.json()
+
+
+def submit_qid_link(qid, case_id, case_api_url):
+    logger.debug('Attempting to submit qid link', qid=qid, case_id=case_id)
+    payload = {'caseId': case_id, 'questionnaireId': qid}
+    response = requests.put(f'{case_api_url}/qids/link', json=payload)
```
Comment (davidmort) [inline census_rm_ops_ui/controllers/case_controller.py:60]: Should be creating and passing a transaction Id to the API here, and we should be logging that transactionid out too along with the audit log (see @NickGrantONS comment)

**Comment by ryangrundy7** [comment]
> Requires some audit logging that can be easily extracted/searched in the logs

@NickGrantONS  The app does have audit logging for everything request that happens 
```
{"audit": true, "requestPath": "/case-details/link-qid/submit/?", "requestMethod": "POST", "user": "ryan.grundy@census.gov.uk", "level": "info", "severity": "info", "service": "census-rm-ops-ui", "created_at": "2020-09-15T07:32:36.440 │
│ 897"}
```

**Comment by NickGrantX** [comment]
> > Requires some audit logging that can be easily extracted/searched in the logs
> 
> @NickGrantONS The app does have audit logging for everything request that happens
> 
> ```
> {"audit": true, "requestPath": "/case-details/link-qid/submit/?", "requestMethod": "POST", "user": "ryan.grundy@census.gov.uk", "level": "info", "severity": "info", "service": "census-rm-ops-ui", "created_at": "2020-09-15T07:32:36.440 │
> │ 897"}
> ```

Needs the transaction ID of any messages which were emitted on a user's behalf, so that we have end-to-end audit trail

**Review by liamtoozer — APPROVED** [review]
LGTM 👍

**Review by NickGrantX — APPROVED** [review]

**Pull request merged**